### PR TITLE
Vm file and codegen added

### PR DESCRIPTION
**Pull request by Mirageinvo**

### CODE REVIEW

**Comment by Mirageinvo** [comment]
@includealex PTAL

File: runtime/interpreter.cpp [hunk 1]
```diff
@@ -33,12 +33,13 @@ VMInstr Interpreter::FetchNext(VMByte *code, VMReg pc)
...
 
-void Interpreter::Run(VMByte *code)
+void Interpreter::Run()
 {
+    VMByte *code = vm_file_->GetCode();
     static void *dispatch_table[] = {&&exit,   &&iadd, &&isub, &&imul, &&idiv, &&fadd, &&fsub,  &&fmul,  &&fdiv,
                                      &&and_,   &&or_,  &&xor_, &&eq,   &&ne,   &&lt,   &&le,    &&gt,    &&ge,
                                      &&feq,    &&fne,  &&flt,  &&fle,  &&fgt,  &&fge,  &&iscan, &&fscan, &&iprint,
-                                     &&fprint, &&sin,  &&cos,  &&pow,  &&sqrt, &&load, &&move,  &&fload, &&fmove};
+                                     &&fprint, &&sin,  &&cos,  &&pow,  &&load, &&move, &&fload, &&fmove};
```
Comment (includealex) [inline runtime/interpreter.cpp:42]: why does `sqrt` disappear?

File: runtime/tests/interpreter_tests.cpp [hunk 2]
```diff
@@ -1,11 +1,11 @@
 #include <gtest/gtest.h>
 #include "interpreter.hpp"
-#include "inst_creator_inl.hpp"
+#include "instr_opcode.gen.hpp"
 
 namespace vm {
 TEST(InterpreterTest, InterpreterTestInitial)
 {
-    Interpreter *inter = Interpreter::CreateInterpreter();
+    Interpreter *inter = Interpreter::CreateInterpreter(nullptr);
```
Comment (includealex) [inline runtime/tests/interpreter_tests.cpp:8]: do we need this change?

**Review by includealex — CHANGES_REQUESTED** [review]

File: runtime/tests/interpreter_tests.cpp (hunk 2 above)
Comment (Mirageinvo) [reply runtime/tests/interpreter_tests.cpp:8]: Currently, interpreter requires VMFile pointer in constructor

File: runtime/interpreter.cpp (hunk 1 above)
Comment (Mirageinvo) [reply runtime/interpreter.cpp:42]: As far as I know it was a workaround because we didn't have working pow for 0.5
Now we have it

Comment (includealex) [reply runtime/interpreter.cpp:42]: Let's leave this one because we have first task with sqrt function.

Comment (Mirageinvo) [reply runtime/interpreter.cpp:42]: We use pow(0.5) for that. Will look like copy paste if we don't remove it

Comment (Mirageinvo) [reply runtime/interpreter.cpp:42]: @includealex

Comment (includealex) [reply runtime/interpreter.cpp:42]: Ok, @Mirageinvo

**Pull request merged**